### ISSUE TITLE
Error: missing server host

### ISSUE THREAD

**Issue by miguelbernadi** [appleboy/ssh-action#300]
I'm consistently getting this error in one workflow, while in another repo it works without issue. The repo where it works is a public repo, while the one where it fails is a private one.

The failing action is:
```
name: Production-CD

on:
  workflow_dispatch:
  push:
    tags:
      - 'v*'

# We ned `contents` to interact with release
permissions:
  contents: write

jobs:
  draft-release:
    runs-on: ubuntu-latest
    outputs:
      upload_url: ${{ steps.create_release.outputs.upload_url }}
      id: ${{ steps.create_release.outputs.id }}
    steps:
    - name: draft-release
      id: create_release
      uses: actions/create-release@v1
      env:
        GITHUB_TOKEN: ${{ github.token }}
      with:
        draft: true
        prerelease: false
        release_name: ${{ github.ref }}
        tag_name: ${{ github.ref }}

  bundle-client:
    runs-on: ubuntu-latest
    needs: [draft-release]
    defaults:
      run:
        working-directory: client

    steps:
      - name: Checkout code
        uses: actions/checkout@v4

      - name: cache node_modules
        id: cache-node-modules
        uses: actions/cache@v4
        with:
          path: '**/node_modules'
          key: ${{ runner.os }}-modules-${{ hashFiles('**/package-lock.json') }}

      - name: Use Node.js 16.8
        uses: actions/setup-node@v4
        with:
          node-version: 16.8

      - name: install dependencies
        if: steps.cache-node-modules.outputs.cache-hit != 'true'
        run: npm ci

      - name: build FE
        run: npm run build

      - run: tar czf crisolAdmin-assets.tar.gz -C build/ .

      - name: upload artifacts
        uses: actions/upload-release-asset@v1.0.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ needs.draft-release.outputs.upload_url }}
          asset_path: ./client/crisolAdmin-assets.tar.gz
          asset_name: crisolAdmin-assets.tar.gz
          asset_content_type: application/gzip

  bundle-server:
    runs-on: ubuntu-latest
    needs: [draft-release]
    defaults:
      run:
        working-directory: server
    steps:
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Use Go 1.18
        uses: actions/setup-go@v4
        with:
          go-version: '^1.18.0'
          cache-dependency-path: server/go.sum

      - run: CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -ldflags '-extldflags "-static"' -ldflags="-X main.Commit=$(git rev-parse HEAD)" -o crisolAdmin

      - name: upload artifacts
        uses: actions/upload-release-asset@v1.0.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ needs.draft-release.outputs.upload_url }}
          asset_path: ./server/crisolAdmin
          asset_name: crisolAdmin
          asset_content_type: application/binary

  publish-release:
    runs-on: ubuntu-latest
    needs:
      - draft-release
      - bundle-client
      - bundle-server
    steps:
      - uses: eregon/publish-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          release_id: ${{ needs.draft-release.outputs.id }}

  deploy:
    runs-on: ubuntu-latest
    needs:
      - publish-release
    permissions:
      actions: write
    concurrency:
      group: ${{ github.workflow }}-${{ github.ref }}
      cancel-in-progress: true
    steps:
      - name: 🚀 Deploy website
        uses: appleboy/ssh-action@v1.0.3
        env:
         TAG: ${{ github.ref_name }}
        with:
          host: ${{ env.DEPLOY_HOST }}
          username: ${{ secrets.DEPLOY_USERNAME }}
          key: ${{ secrets.CRISOL_ADMIN_DEPLOY_PRIVATE_KEY }}
          port: ${{ secrets.DEPLOY_PORT }}
          command_timeout: 20m
          script: TAG=$TAG ./deploy-tag.sh
```
Any ideas on what may be the issue or how to debug it?

**Comment by appleboy** [comment]
What is your `${{ env.DEPLOY_HOST }}` value?

![image](https://github.com/appleboy/ssh-action/assets/21979/66981735-84f2-474d-8e1f-99fcea51f86a)

**Comment by ariwijayaikd** [comment]
> What is your `${{ env.DEPLOY_HOST }}` value?
> 
> ![image](https://private-user-images.githubusercontent.com/21979/310021761-66981735-84f2-474d-8e1f-99fcea51f86a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2MjU1NTMsIm5iZiI6MTcwOTYyNTI1MywicGF0aCI6Ii8yMTk3OS8zMTAwMjE3NjEtNjY5ODE3MzUtODRmMi00NzRkLThlMWYtOTlmY2VhNTFmODZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA1VDA3NTQxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjOTQxZTE5YWMzOTk2NzVmOGU3MzcyNjVhMWU5NDdlYjFjY2FmYzNjMTQ5ODVmMmExMWIwNWQ2ZjllMGU3OGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.191PO1E4mg1h1g6iBjB_nKURtUnDmOnjsKlVhaYrpXg)

I have same issue, the value was variable, everything was normal until yesterday
![image](https://github.com/appleboy/ssh-action/assets/80755160/3d603007-255d-4fdd-b374-9aff10875b80)
![image](https://github.com/appleboy/ssh-action/assets/80755160/6a94aef0-ba83-412a-854a-058eac7b6ab5)

**Comment by ariwijayaikd** [comment]
I just change the runner using github action runner in private vm (internal network) and it work

**Comment by appleboy** [comment]
@ariwijayaikd OK. I think it is a firewall issue. Thanks for your feedback

**Comment by miguelbernadi** [comment]
The contents of  `${{ env.DEPLOY_HOST }}` were the DNS domain I was targeting. I hardcoded it instead of using a variable, and now it works. I also had to change the secrets I was using as I discovered organization secrets are not used in actions in our current paying tier. I suspect it may also complain about the host when the key is empty, but I cannot confirm it.

As I managed to solve it by avoiding variables and using repository secrets, I no longer have an issue, but error messages may be misleading.
